### PR TITLE
Disallow relayMessage to L2 to L1 Message Passer

### PR DESCRIPTION
**Pull request by maurelian**
This change prevents call spoofing on L2.
Adds a test based on logs and successfulMessages mapping.

### CODE REVIEW

File: contracts/optimistic-ethereum/OVM/bridge/messaging/OVM_L1CrossDomainMessenger.sol [hunk 1]
```diff
@@ -19,9 +19,9 @@ import { Abs_BaseCrossDomainMessenger } from "./Abs_BaseCrossDomainMessenger.sol
 
 /**
  * @title OVM_L1CrossDomainMessenger
- * @dev The L1 Cross Domain Messenger contract sends messages from L1 to L2, and relays messages from L2 onto L1. 
- * In the event that a message sent from L1 to L2 is rejected for exceeding the L2 epoch gas limit, it can be resubmitted 
- * via this contract's replay function. 
+ * @dev The L1 Cross Domain Messenger contract sends messages from L1 to L2, and relays messages from L2 onto L1.
+ * In the event that a message sent from L1 to L2 is rejected for exceeding the L2 epoch gas limit, it can be resubmitted
+ * via this contract's replay function.
```
Comment (maurelian) [inline contracts/optimistic-ethereum/OVM/bridge/messaging/OVM_L1CrossDomainMessenger.sol:24]: Lots of whitespace fixes in this PR from the recently added editorConfig file.

File: contracts/optimistic-ethereum/OVM/bridge/messaging/OVM_L2CrossDomainMessenger.sol [hunk 2]
```diff
@@ -75,6 +75,15 @@ contract OVM_L2CrossDomainMessenger is iOVM_L2CrossDomainMessenger, Abs_BaseCros
             "Provided message has already been received."
         );
 
+        // Prevent calls to OVM_L2ToL1MessagePasser, which would enable
+        // an attacker to maliciously craft the _message to spoof
+        // a call from any L2 account.
+        if(_target == resolve("OVM_L2ToL1MessagePasser")){
+            // Write to the successfulMessages mapping and return immediately.
+            successfulMessages[xDomainCalldataHash] = true;
```
Comment (maurelian) [inline contracts/optimistic-ethereum/OVM/bridge/messaging/OVM_L2CrossDomainMessenger.sol:83]: One could argue against calling this a 'successful message', but I think preventing future attempts to relay it is the right move.

File: test/contracts/OVM/bridge/base/OVM_L2CrossDomainMessenger.spec.ts [hunk 3]
```diff
@@ -157,5 +158,40 @@ describe('OVM_L2CrossDomainMessenger', () => {
...
+        resProm
+      ).to.not.be.reverted
+
+      // There should be no 'relayedMessage' event logged in the receipt.
+      const logs = (
+        await Mock__OVM_L2ToL1MessagePasser.provider.getTransactionReceipt(
+          (await resProm).hash
+        )
+      ).logs
+      expect(logs).to.deep.equal([])
```
Comment (maurelian) [inline test/contracts/OVM/bridge/base/OVM_L2CrossDomainMessenger.spec.ts:184]: This check is an indirect way of verifying that "no call was made to the Message Passer", I would have preferred to use `Mock__OVM_L2ToL1MessagePasser.smocked.passMessageToL1.calls.length === 0`, but smock doesn't actually handle the case where no calls are made to a function. I spend some time on a fix, but other tests broke, so I went with this.

File: contracts/optimistic-ethereum/OVM/bridge/messaging/OVM_L2CrossDomainMessenger.sol (hunk 2 above)
Comment (ben-chain) [reply contracts/optimistic-ethereum/OVM/bridge/messaging/OVM_L2CrossDomainMessenger.sol:83]: Agreed, nice one!

**Review by ben-chain — APPROVED** [review]
This LGTM!

**Pull request merged**